src/auth/src/credentials/service_account_credential.rs

```diff
@@ -82,7 +82,21 @@ struct ServiceAccountTokenProvider {
 #[async_trait]
 impl TokenProvider for ServiceAccountTokenProvider {
     async fn get_token(&self) -> Result<Token> {
-        let signer = self.signer(&self.service_account_info.private_key)?;
+        let signing_key = self.get_signing_key(&self.service_account_info.private_key)?;
+        let signing_algorithm = match signing_key.algorithm().as_str().unwrap() {
+            "RSA" => Ok("RS256"),
+            "ECDSA" => Ok("ES256"),
+            alg => Err(CredentialError::non_retryable_from_str(format!(
+                "Unsupported signing algorithm {} ",
+                alg
+            ))),
```

Comment:
nit: trailing space in error message, also consider:

```suggestion
            alg => Err(CredentialError::non_retryable_from_str(format!(
                "Unsupported signing algorithm {alg}"))),
```